feat: `exec`, `run`, `info strict` commands implementation, flags renaming. (#3723)
gruntwork-io/terragrunt: docs/_docs/04_reference/02-cli-options.md

- If any unit throws a 2, but nothing throws a 1, Terragrunt will throw a 2.
- If nothing throws a non-zero, Terragrunt will throw a 0.

#### graph

Run the provided OpenTofu/Terraform command against the graph of dependencies for the unit in the current working directory. The graph consists of all units that depend on the unit in the current working directory via a `dependency` or `dependencies` blocks, plus all the units that depend on those units, and all the units that depend on those units, and so on, recursively up the tree, up to the Git repository root, or the path specified via the optional `--graph-root` argument.

> The Command will be executed following the order of dependencies: so it'll run on the unit in the current working directory first, then on units that depend on it directly, then on the units that depend on those units, and so on. Note that if the command is `destroy`, it will execute in the opposite order of the dependencies.

Reviewer: ```suggestion
The Command will be executed following the order of dependencies: so it'll run on the unit in the current working directory first, then on units that depend on it directly, then on the units that depend on those units, and so on. Note that if the command is `destroy`, it will run in the opposite order of the dependencies (the final dependents, then their dependencies, etc. up to the unit you ran the command in).
```

  path   = "app1"
}

unit "app2" {
  source = "github.com/gruntwork-io/terragrunt.git//test/fixtures/inputs?ref=${local.version}"
  path   = "app2"
}

```

Executing generate:

```bash
terragrunt stack generate
```

Will create the following directory structure:

```tree
.stack/
├── app1/
│   └── terragrunt.hcl
└── app2/
    └── terragrunt.hcl
```

#### stack run

The `stack run *` command allows users to execute IaC commands across all units defined in a `terragrunt.stack.hcl` file.
This feature facilitates efficient orchestration of operations on multiple units, simplifying workflows for managing complex infrastructure stacks.

**Examples:**

Run a plan on each unit:

```bash
terragrunt stack run plan
```

Apply changes for each unit:

```bash
terragrunt stack run apply
```

Destroy all units:

```bash
terragrunt stack run destroy
```

**Note:**

Before executing the specified command, the `terragrunt stack run *` command will automatically generate the stack by creating
the `.terragrunt-stack` directory using the `terragrunt.stack.hcl` configuration file.
This ensures that all units are up-to-date before running the requested operation.

### info

> Emits limited terragrunt state on `stdout` in a JSON format and exits.

Reviewer: ```suggestion
Emits limited terragrunt information to stdout in JSON format.
```
